modules/mongodb/src/main/java/org/testcontainers/containers/MongoDBContainer.java

```diff
@@ -87,7 +92,8 @@ protected void containerIsStarted(InspectContainerResponse containerInfo) {
     }
 
     private String[] buildMongoEvalCommand(final String command) {
-        return new String[] { "mongo", "--eval", command };
+        String cmd = this.isAtLeastVersion6 ? "mongosh" : "mongo";
+        return new String[] { cmd, "--eval", command };
```

Comment:
@eddumelendez WDYT about not using `ComparableVersion` but doing "mongosh || mongo"-style command, so that the shell will "fallback" automatically?

Reply:
That sounds like a good idea, let's do a follow-up PR for this @eddumelendez and keep this in mind as a pattern when possible 👍